pkg/scalers/metrics_api_scaler.go

```diff
@@ -13,20 +13,59 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/metrics/pkg/apis/external_metrics"
 	"net/http"
+	neturl "net/url"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"strconv"
+	"strings"
+	"time"
 )
 
 type metricsAPIScaler struct {
 	metadata *metricsAPIScalerMetadata
+	client   *http.Client
 }
 
 type metricsAPIScalerMetadata struct {
 	targetValue   int
 	url           string
 	valueLocation string
+
+	//apiKeyAuth
+	enableAPIKeyAuth bool
+	// +default is header
+	method method
+	// +option default header key is X-API-KEY and default query key is api_key
+	keyParamName string
+	apiKey       string
+
+	//base auth
+	enableBaseAuth bool
+	username       string
+	// +optional
+	password string
+
+	//client certification
+	enableTLS bool
+	cert      string
+	key       string
+	ca        string
 }
 
+type authenticationType string
+
+const (
+	apiKeyAuth authenticationType = "apiKeyAuth"
+	basicAuth                     = "basicAuth"
+	tlsAuth                       = "tlsAuth"
+)
+
+type method string
+
+const (
+	header     method = "header"
+	queryParam        = "query"
```

Comment:
```suggestion
	headerMethod     method = "header"
	queryParamMethod        = "query"
```
How about this?  I think this may introduce some clarity

Reply:
I have removed this section altogether. This will now be part of the documentation itself.